src/mlpack/methods/ne/tasks.hpp

```diff
@@ -0,0 +1,186 @@
+    double theta_limit = 12 * M_PI / 180.0;
+    double g = 9.81;
+    double mp = 0.1;
+    double mc = 1.0;
+    double l = 0.5;
+    double F = 10.0;
+    double tau = 0.02;
+    ssize_t num_trial = 20;
+    ssize_t num_step = 200;
+  }
+
+  // Parametric constructor.
+  TaskCartPole(double t_mc, double t_mp, double t_g, double t_l, double t_F, 
+               double t_tau, double t_track_limit, double t_theta_limit,
+               ssize_t t_num_trial, ssize_t t_num_step):
+               mc(t_mc), mp(t_mp), g(t_g), l(t_l), F(t_F),
+               tau(t_tau), track_limit(t_track_limit), theta_limit(t_theta_limit),
+               num_trial(t_num_trial), num_step(t_num_step) {}
+
+  // Update status.
+  void Action(double action, double& x, double& x_dot, double& theta, double& theta_dot) {
+    double force = action * F;  // action is -1 or 1. TODO: or continuous???
+    double cos_theta = cos(theta);
+    double sin_theta = sin(theta);
+    double total_mass = mp + mc;
+    double temp = (force + mp * l * theta_dot * theta_dot * sin_theta) / total_mass;   
+    double theta_acc = (g * sin_theta - cos_theta * temp) / (l * (4.0 / 3.0 - mp * cos_theta * cos_theta / total_mass));    
+    double x_acc  = temp - mp * l * theta_acc * cos_theta / total_mass;
+
+    // Update the four state variables using Euler's method.
+    x += tau * x_dot;
+    x_dot += tau * x_acc;
+    theta += tau * theta_dot;
+    theta_dot += tau * theta_acc;
+  }
+
+  // Evaluate genome's fitness for this task.
+  double EvalFitness(Genome& genome) {
+    assert(genome.NumInput() == 5); 
+    assert(genome.NumOutput() == 1);
+
+    mlpack::math::RandomSeed(1);  // If no seed set, each time the fitness will change.
+    double fitness = 0;
+    for (ssize_t trial=0; trial<num_trial; ++trial) {
+      // Initialize inputs: x, x_dot, theta, theta_dot. As used by Stanley
+      double x = mlpack::math::Random(-2.4, 2.4);
+      double x_dot = mlpack::math::Random(-1.0, 1.0);
+      double theta = mlpack::math::Random(-0.2, 0.2);
+      double theta_dot = mlpack::math::Random(-1.5, 1.5);
+
+      for (ssize_t step=0; step<num_step; ++step) {
+        // Convert input into range [0, 1]
+        std::vector<double> inputs = {(x + 2.4) / 4.8,
+                                      (x_dot + 0.75) / 1.5,  // ??
+                                      (theta + theta_limit) / 0.41,
+                                      (theta_dot + 1.0) / 2.0,
+                                      1};  // TODO: use arma::mat for input.
+        genome.Activate(inputs);
+        std::vector<double> output;
+        genome.Output(output);
+
+        double action = output[0];
+        if (output[0] < 0.5) {
+          action = -1;
+        } else {
+          action = 1;
+        }
+        
+        Action(action, x, x_dot, theta, theta_dot);
+        fitness += 1;
+        if (abs(x)>=track_limit || abs(theta)>=theta_limit) break;
+      }
+    }
+
+    return (1.0 / fitness);
```

Comment:
And we can let algorithm stop when fitness reaches to 0, for both XOR and CartPole.
Do you think it is a good idea?